### DISCUSSION TITLE
Os::Task Updates

### DISCUSSION THREAD

**Discussion by LeStarch** [nasa/fprime#1041] [Announcements]
Community,

I am about to merge a change to the `devel` branch, which will affect the `Os::Task.starct()` and `<active component>.start()` interfaces.  It is recommended you take a moment, if using `devel`, to update your start calls.

1. If you don't care about priority (or run without permissions to set priority), switch to calling `.start()` (no arguments) in your topology.
2. If you use priority, please re-write `.start()` calls to use the ordering `.start(priority, stack size)`
3. **If you use CPU affinity** you must switch to the following interface
```
.start(static_cast<NATIVE_UINT_TYPE>(priority), static_cast<NATIVE_UINT_TYPE>(stackSize), static_cast<NATIVE_UINT_TYPE>(cpu affinity));
``` 
This is to disambiguate the new API call with priority first from the old interface (superfluous thread id first).

The new API arguments are all optional, in the order:  priority, stack size, affinity, identifier.

**It is strongly recommended** that you respond to any deprecated warnings you get when compiling.  These warnings will become errors in the next release.   The full work can be seen here: https://github.com/nasa/fprime/pull/992

**Comment by ThibFrgsGmz** [comment]
Given the importance of the announcement, shouldn't it be pinned for a while in the Discussions tab so that it doesn't get drowned out with the other discussions?

**Reply by LeStarch** [reply]
Done.  Didn't realize one can pin announcements, thanks for the tip!
